modules/gateway/outputs.tf

```diff
@@ -24,10 +24,32 @@ output "public_ip_address_fqdn" {
 
 output "gateway_id" {
   description = "Identifier of the application gateway for GraphDB"
-  value       = var.gateway_enable_private_access ? azurerm_application_gateway.graphdb-private[0].id : azurerm_application_gateway.graphdb-public[0].id
+  value       = var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-private) > 0 ? azurerm_application_gateway.graphdb-private[0].id : !var.gateway_enable_private_access && length(azurerm_application_gateway.graphdb-public) > 0 ? azurerm_application_gateway.graphdb-public[0].id : null
 }
 
+# Gateway Backend Adress Pool ID:
+# Check if application gateway is enabled and private access is true. If both are available, checks for backend adress pool and retrieves the ID of the first one. Otherwise returns Null.
```

Comment:
Checks if the Application Gateway and Private access are enabled, if true then retrieves the first backend address pool ID, if false return null.

Reply:
sounds better. Changed